### PR TITLE
CodeQL docs migration prep: Update QL tutorials

### PR DESCRIPTION
**Pull request by shati-patel**
## Summary

This rearranges and updates the "QL tutorials" map topic according to the docs pre-migration issue [2173](https://github.com/github/product-documentation/issues/2173). 

The reshuffling involved quite a lot of changes already, so I haven't made any updates to the content yet (apart from adding intros).

### General points

- I struggled to come up with good introductions for these articles. The "tutorial" content type doesn't exist yet in the GitHub docs, so I wasn't sure what form to use... Please suggest changes if you have better ideas!
- There are a few changes to update other files/TOCs, but these are minor and not included in the overview table below.

### Changes in this PR

| Original | New preview | Source | Notes |
| --- | --- | --- | --- |
| [QL detective tutorials](https://help.semmle.com/QL/learn-ql/beginner/ql-tutorials.html) | [QL tutorials](http://docteam.internal.semmle.com/shati/learn-ql/beginner/ql-tutorials.html) | [Source](https://github.com/Semmle/ql/pull/3043/files#diff-d007609a67dbffe2a3cc8d4a45316ee0) | |
| Find the thief pages [1](https://help.semmle.com/QL/learn-ql/beginner/find-thief-1.html), [2](https://help.semmle.com/QL/learn-ql/beginner/find-thief-2.html), and [3](https://help.semmle.com/QL/learn-ql/beginner/find-thief-3.html) | [Find the thief](http://docteam.internal.semmle.com/shati/learn-ql/beginner/find-the-thief.html) | [Source](https://github.com/Semmle/ql/pull/3043/files#diff-867d123f4c9433ab2a8505af98c4fb4b) | Combined 3 pages, updated some headers, and added intro |
| Catch the fire starter pages [1](https://help.semmle.com/QL/learn-ql/beginner/fire-1.html) and [2](https://help.semmle.com/QL/learn-ql/beginner/fire-2.html) | [Catch the fire starter](http://docteam.internal.semmle.com/shati/learn-ql/beginner/catch-the-fire-starter.html) | [Source ](https://github.com/Semmle/ql/pull/3043/files#diff-6517c3c07b464ef5f6a6d40f8a8753ff)| Combined 2 pages, updated some headers, and added intro |
| [Crown the rightful heir](https://help.semmle.com/QL/learn-ql/beginner/heir.html) | [Crown the rightful heir](http://docteam.internal.semmle.com/shati/learn-ql/beginner/crown-the-rightful-heir.html) | [Source ](https://github.com/Semmle/ql/pull/3043/files#diff-684eb72498d756bc13f287e5728e22cd)| Added intro and moved header |
| [River crossing puzzle](https://help.semmle.com/QL/learn-ql/ql-etudes/river-crossing.html) | [Cross the river](http://docteam.internal.semmle.com/shati/learn-ql/beginner/cross-the-river.html) | [Source ](https://github.com/Semmle/ql/pull/3043/files#diff-bddaced5ccd4c8b84fbd24e96a8f6ea2)| Moved into new location, changed title, and added intro |

### CODE REVIEW

File: docs/language/learn-ql/beginner/ql-tutorials.rst [hunk 1]
```diff
@@ -1,27 +1,14 @@
...
-Currently the following detective tutorials are available:
-
--  :doc:`Find the thief <find-thief-1>`—a three part mystery that introduces logical connectives, quantifiers, and aggregates
--  :doc:`Catch the fire starter <fire-1>`—an intriguing search that introduces predicates and classes
--  :doc:`Crown the rightful heir <heir>`—a detective puzzle that introduces recursion
-
-Further resources
------------------
+.. toctree::
+   :glob:
```
Comment (jf205) [inline docs/language/learn-ql/beginner/ql-tutorials.rst]:
```suggestion
```

Not needed (and it seems to muck up the ordering in the sidebar TOC 🤷‍♂ )

File: docs/language/learn-ql/beginner/crown-the-rightful-heir.rst [hunk 2]
```diff
@@ -1,6 +1,11 @@
 Crown the rightful heir
 =======================
 
+This is QL detective puzzle that shows you how to use recursion in QL to write more complex queries.
```
Comment (jf205) [inline docs/language/learn-ql/beginner/crown-the-rightful-heir.rst]:
```suggestion
This is a QL detective puzzle that shows you how to use recursion in QL to write more complex queries.
```

File: docs/language/learn-ql/beginner/find-the-thief.rst [hunk 3]
```diff
@@ -0,0 +1,297 @@
+Find the thief
+==============
+
+Take on the role of a detective to find the thief in this fictional village. You will learn how to use logical connectives, quantifiers, and aggregates in QL along the way.
```
Comment (felicitymay) [inline docs/language/learn-ql/beginner/find-the-thief.rst]: Love this introduction 💛

File: docs/language/learn-ql/beginner/ql-tutorials.rst [hunk 4]
```diff
@@ -1,27 +1,14 @@
...
-.. toctree::
-   :glob:
-   :hidden:
-
-   ./*
-
-Welcome to the detective tutorials! These are aimed at complete beginners who would like to learn the basics of QL,
-before analyzing code with CodeQL.
-The tutorials teach you how to write queries and introduce you to key logic concepts along the way.
+Solve puzzles to learn the basics of QL before analyzing code with CodeQL. The tutorials teach you how to write queries and introduce you to key logic concepts along the way.
```
Comment (felicitymay) [inline docs/language/learn-ql/beginner/ql-tutorials.rst]: Possibly worth making this change from the localization point of view:

```suggestion
Solve puzzles to learn the basics of QL before you analyze code with CodeQL. The tutorials teach you how to write queries and introduce you to key logic concepts along the way.
```

File: docs/language/learn-ql/index.rst [hunk 5]
```diff
@@ -34,7 +34,6 @@ If you are new to QL, start by looking at the following topics:
    introduction-to-ql
    about-ql
    beginner/ql-tutorials
-   ql-etudes/river-crossing
```
Comment (felicitymay) [inline docs/language/learn-ql/index.rst]: Does this topic need an introduction too?

Comment (shati-patel) [reply docs/language/learn-ql/index.rst]: This topic won't exist in the migrated help (though we can use some of the content in https://github.com/github/product-documentation/issues/2159). I've just updated the TOC so it doesn't use a broken link!

**Comment by shati-patel** [comment]
Thank you for having a look! 🔍 
I've made some changes to the intros according to your suggestions.

**Pull request merged**